src/components/common/Dialogs/EditDomainDialog/EditDomainDialogForm.tsx

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import { ColonyRole } from '@colony/colony-js';
+import { defineMessages } from 'react-intl';
+import { useFormContext } from 'react-hook-form';
+
+import {
+  ActionDialogProps,
+  DialogControls,
+  DialogHeading,
+  DialogSection,
+} from '~shared/Dialog';
+import { HookFormInput as Input, Annotations } from '~shared/Fields';
+import { getDomainOptions } from '~shared/DomainFundSelectorSection/helpers';
+import NoPermissionMessage from '~shared/NoPermissionMessage';
+import PermissionRequiredInfo from '~shared/PermissionRequiredInfo';
+import DomainNameAndColorInputGroup from '~shared/DomainNameAndColorInputGroup';
+import CannotCreateMotionMessage from '~shared/CannotCreateMotionMessage';
+// import NotEnoughReputation from '~dashboard/NotEnoughReputation';
+
+import { useDialogActionPermissions, useEnabledExtensions } from '~hooks';
+import { DomainColor } from '~gql';
+import { notNull } from '~utils/arrays';
+import { findDomainByNativeId } from '~utils/domains';
+import { noMotionsVotingReputationVersion } from '~utils/colonyMotions';
+
+const displayName = 'common.EditDomainDialog.EditDomainDialogForm';
+
+const MSG = defineMessages({
+  titleEdit: {
+    id: `${displayName}.titleEdit`,
+    defaultMessage: 'Edit team details',
+  },
+  name: {
+    id: `${displayName}.name`,
+    defaultMessage: 'Team name',
+  },
+  purpose: {
+    id: `${displayName}.purpose`,
+    defaultMessage: 'What is the purpose of this team?',
+  },
+  annotation: {
+    id: `${displayName}.annotation`,
+    defaultMessage: 'Explain why you’re editing this team (optional)',
+  },
+  cannotCreateMotion: {
+    id: `${displayName}.cannotCreateMotion`,
+    defaultMessage: `Cannot create motions using the Governance v{version} Extension. Please upgrade to a newer version (when available)`,
+  },
+});
+
+const EditDomainDialogForm = ({
+  back,
+  colony,
+  colony: { domains },
+}: ActionDialogProps) => {
+  const {
```

Comment:
Same thought here re wrapping this logic into a hook (applies to all components really where there's more than a few lines of business logic above the view (so I don't repeat myself any more in the comments))